streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -112,6 +115,11 @@ public File baseDir() {
      * @throws StreamsException if the store's change log does not contain the partition
      */
     public void register(StateStore store, boolean loggingEnabled, StateRestoreCallback stateRestoreCallback) {
+        Map<String, String> sourceStoreToSourceTopic = null;
+        if (topology != null) {
```

Comment:
Will topology ever be `null`? i.e., is it valid for it to be `null`? if not then this could be simplified